src/hotspot/share/gc/g1/g1ParScanThreadState.hpp

```diff
@@ -130,28 +128,16 @@ class G1ParScanThreadState : public CHeapObj<mtGC> {
 
   void push_on_queue(ScannerTask task);
 
-  template <class T> void enqueue_card_if_tracked(G1HeapRegionAttr region_attr, T* p, oop o) {
-    assert(!HeapRegion::is_in_same_region(p, o), "Should have filtered out cross-region references already.");
-    assert(!_g1h->heap_region_containing(p)->is_young(), "Should have filtered out from-young references already.");
+  // Apply the post barrier to the given reference field. Enqueues the card of p
+  // if the barrier (same region, not from survivor) does not filter out not find
```

Comment:
I don't understand this comment; seems like there's a grammar or wording problem.